Library/Homebrew/formula_installer.rb

```diff
@@ -212,6 +213,10 @@ def install
       opoo "#{formula.full_name}: #{old_flag} was deprecated; using #{new_flag} instead!"
     end
 
+    invalid_flag_names.each do |flag|
+      opoo "#{formula.full_name}: #{flag} is invalid for this formula and will be ignored!"
```

Comment:
I think on typos it's worth just hard failing. We don't use Levenshtein or anything similar currently but would be 👍 on doing so in future.